### ISSUE TITLE
CloseIndexIT#testCloseWhileIndexingDocuments can fail with assertion error

### ISSUE THREAD

**Issue by jtibshirani** [elastic/elasticsearch#37897]
Unfortunately I'm not able to reproduce this locally. Although this is the first time the test has failed in this way, I wanted to raise an issue because there was a recent, related attempt to fix the test (https://github.com/elastic/elasticsearch/commit/ef8dd12c6d87f40e2c41da274bbb08136ae164e4).

I opted not to mute the test yet, since it's the first time it has failed in this way.

---

Link to the build: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/1560/console

Command to reproduce: 
```
./gradlew :server:integTest \
  -Dtests.seed=6BBE973459D47F76 \
  -Dtests.class=org.elasticsearch.indices.state.CloseIndexIT \
  -Dtests.method="testCloseWhileIndexingDocuments" \
  -Dtests.security.manager=true \
  -Dtests.locale=de-GR \
  -Dtests.timezone=Antarctica/Rothera \
  -Dcompiler.java=11 \
  -Druntime.java=8
```

Relevant excerpt from the logs:
```
FAILURE 2.47s J3 | CloseIndexIT.testCloseWhileIndexingDocuments <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: Count is 25000+ hits but 25000 was expected.  Total shards: 2 Successful shards: 2 & 0 shard failures:
   > 	at __randomizedtesting.SeedInfo.seed([6BBE973459D47F76:DD7559B9E3B005E9]:0)
   > 	at org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount(ElasticsearchAssertions.java:256)
   > 	at org.elasticsearch.indices.state.CloseIndexIT.testCloseWhileIndexingDocuments(CloseIndexIT.java:191)
   > 	at java.lang.Thread.run(Thread.java:748)
```

**Comment by tlrx** [comment]
Closed in 758eb9d451eebd90d787c1e6ec70a49b4dd20c73

Thanks @jtibshirani for reporting this failure, it helped to find #37907. I fixed this specific test in the meanwhile.

**Comment by tlrx** [comment]
And thanks @jimczi of course :)
